src/core/src/consensus_state_store.rs

```diff
@@ -67,6 +68,129 @@ impl<'a> ConsensusStateStore<'a> {
+        let reader = db.reader().unwrap();
+
+        assert!(reader.index_get(index, &[key]).unwrap().is_none());
+    }
+
+    fn create_context() -> Result<lmdb::LmdbContext, CliError> {
+        let path_config = config::get_path_config();
+        let statestore_path = &path_config.data_dir.join(config::get_filename());
+
+        lmdb::LmdbContext::new(statestore_path, 1, None)
+            .map_err(|err| CliError::EnvironmentError(format!("{}", err)))
+    }
+
+    #[test]
+    fn test_state_store_get() {
+
+        let mut ctx = create_context().unwrap();
+        let statestore_db = lmdb::LmdbDatabase::new(
+            &ctx,
+            &["index_consensus_state"],
+        ).map_err(|err| CliError::EnvironmentError(format!("{}", err))).unwrap();
+
+        let mut state_store = ConsensusStateStore::new(statestore_db);
+
+        assert!(state_store.get(BlockId::from(vec![11])).is_err());
+        state_store.put( BlockId::from(vec![11]), ConsensusState::default() );
+
+        assert!(state_store.get(BlockId::from(vec![11])).is_ok());
+        //cleanup
+        state_store.delete( BlockId::from(vec![11]) );
+    }
+
+    #[test]
+    fn test_state_store_put() {
+
+        let mut ctx = create_context().unwrap();
+        let statestore_db = lmdb::LmdbDatabase::new(
+            &ctx,
+            &["index_consensus_state"],
+        ).map_err(|err| CliError::EnvironmentError(format!("{}", err))).unwrap();
+
+        let mut state_store = ConsensusStateStore::new(statestore_db);
+
+        assert!(state_store.get(BlockId::from(vec![13])).is_err());
+        state_store.put( BlockId::from(vec![13]), ConsensusState::default() );
+
+        assert!(state_store.get(BlockId::from(vec![13])).is_ok());
+        assert_eq!(*state_store.get(BlockId::from(vec![13])).unwrap(),
+            ConsensusState::default());
+        //cleanup
+        state_store.delete( BlockId::from(vec![13]) );
+    }
+
+    #[test]
+    fn test_state_store_delete() {
+
+        let mut ctx = create_context().unwrap();
+        let statestore_db = lmdb::LmdbDatabase::new(
+            &ctx,
+            &["index_consensus_state"],
+        ).map_err(|err| CliError::EnvironmentError(format!("{}", err))).unwrap();
+
+        let mut state_store = ConsensusStateStore::new(statestore_db);
+
+        state_store.put( BlockId::from(vec![14]), ConsensusState::default() );
+        state_store.put( BlockId::from(vec![15]), ConsensusState::default() );
+
+        assert_eq!(*state_store.get(BlockId::from(vec![14])).unwrap(),
+            ConsensusState::default());
+        assert_eq!(*state_store.get(BlockId::from(vec![15])).unwrap(),
+            ConsensusState::default());
+
+        state_store.delete( BlockId::from(vec![14]) );
+        assert!(state_store.get(BlockId::from(vec![14])).is_err());
+        assert!(state_store.get(BlockId::from(vec![15])).is_ok());
+        //cleanup
+        state_store.delete( BlockId::from(vec![15]) );
```

Comment:
You could add  assert!(state_store.get(BlockId::from(vec![15])).is_err());

Reply:
This last deletion was meant only for cleanup of the db after actual test run. Will put an assert anyways.